### PR TITLE
use go-sysinfo v1.14.1 and elastic-agent-system-metrics v0.10.6

### PR DESCRIPTION
**Pull request by AndersonQ**
## Proposed commit message

use go-sysinfo v1.14.1 and elastic-agent-system-metrics v0.10.6

go-sysinfo was lower-casing FQDN by default, this new version does not. Not its user should do it if necessary. elastic-agent-system-metrics is also upgraded to a version that lowercase the FQDN when it's used as `host.name`

## Checklist

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Disruptive User Impact

It might cause the FQDN or anything using it to be reported with mixed case instead of lower case

## How to test this PR locally

The same idea as described on https://github.com/elastic/beats/issues/39993
Ingest data, ensure the host metadata is added and check them on kibana

- Closes https://github.com/elastic/beats/issues/39993

## Screenshots

Screenshot from 2024-07-23 18-11-13

### CODE REVIEW

**Comment by VihasMakwana** [comment]
@AndersonQ I think the test cases are failing because the non-fatal PR is yet to be merged. https://github.com/elastic/beats/pull/40298, it requires one more review from code owner and we should be good to go.

**Review by jlind23 — APPROVED** [review]

**Comment by jlind23** [comment]
Removed by review, thought it was just a version bump but saw a change in beat.go file.

**Review by VihasMakwana — APPROVED** [review]
LGTM!

**Comment by lucabelluccini** [comment]
For anyone looking at this PR: it was not backported and it will be in 8.16.x as the change might constitute a breaking change in terms of host names

**Pull request merged**
